File: duplicate-zeros/duplicate-zeros.java
class Solution {
    public void duplicateZeros(int[] arr) {
        int c = 0;
        for(int i = 0; i<arr.length; i++)
        {
            if(arr[i] == 0)
                c++;
        }
        int len = arr.length + c;
        int i = arr.length-1;
        int j = len-1;
        while(i<j)
        {
            if(arr[i]!=0)
            {
                if(j<arr.length)
                {
                    arr[j] = arr[i];
                }
            }
            else 
            {
                if(j<arr.length)
                    arr[j] = arr[i];
                j--;
                if(j<arr.length)
                    arr[j] = arr[i];
            }
            i--;
            j--;
        }        
    }
}

File: duplicate-zeros/duplicate-zeros.cpp
class Solution {
public:
    void duplicateZeros(vector<int>& arr) 
    {
        int c = 0;
        for(int i = 0; i<arr.size(); i++)
        {
            if(arr[i] == 0)
                c++;
        }
        int i = arr.size()-1;
        int j = arr.size()+c-1;
        while(i<j)
        {
            if(arr[i] != 0)
            {
                if(j<arr.size())
                {
                    arr[j] = arr[i];
                }
            }
            else 
            {
                if(j<arr.size())
                {
                    arr[j] = arr[i];
                }
                j--;
                 if(j<arr.size())
                {
                    arr[j] = arr[i];
                } 
            }
            i--;
            j--;
        }
    }
};